### PR TITLE
Show `extra` data on the connection list view

### PR DESCRIPTION
**Pull request by yyyyyyuanfei**
Dear Airflow Maintainers,

Please accept the following PR that
- shows the `extra` of each connection in a neat way:
  image
  - adds a column in the last to indicate whether the connection has extra. 
  - if it does, show a plus sign where the user can click to toggle the detail view.

This PR is related to the corresponding [roadmap item](https://github.com/airbnb/airflow/wiki/Roadmap#ui), I'd like to continue working on the edit form component if necessary. 

The same goes to DagRun `conf` property, do you feel it's good to expose that as well? We internally have implemented that column in the DAG run view, it's quite helpful.

Reminder to contributors:
- You must add an Apache License header to all new files
- Please squash your commits when possible and follow the [7 rules of good Git commits](http://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: airflow/www/static/footable.core.css [hunk 1]
```diff
@@ -0,0 +1,218 @@
+@font-face {
```
Comment (r39132) [inline airflow/www/static/footable.core.css:15]: Please add an apache license header. All files require one.  Here are examples from another Apache project : https://github.com/apache/falcon/tree/master/html5-ui/css

File: airflow/www/views.py [hunk 2]
```diff
@@ -33,7 +33,6 @@
 import sqlalchemy as sqla
```
Comment (r39132) [inline airflow/www/views.py]: Did you run some sort of autopep8 lint-fixer on this? Looks like a lot of minor line changes that are unrelated to any logic changes needed for this feature. If so, can you undo them? It will make it easier to spot logic-only changes. At some point, we might need to do an autopep8 run on all files, but we will wait for that day.

**Comment by r39132** [comment]
Consider : 
1. squashing your commits
2. Undoing the autoPep8 that seems to have run on the views.py file.
3. Add the missing license header to the first css file in your diff.

File: airflow/www/static/footable.core.css (hunk 1 above)
Comment (yyyyyyuanfei) [inline airflow/www/static/footable.core.css:15]: It's a 3rd party library, is it ok to add apache license? What if the file already has a license of different type?

File: airflow/www/views.py (hunk 2 above)
Comment (yyyyyyuanfei) [inline airflow/www/views.py]: Yes, I'll undo it, thanks for pointing out.

File: airflow/www/static/footable.core.css (hunk 1 above)
Comment (r39132) [inline airflow/www/static/footable.core.css:15]: Oh, I might have misread this. It's MIT, so please put the MIT license header here (not the Apache one -- that is not allowed). 

Then, please update NOTICE.txt : follow https://github.com/apache/incubator-ranger/blob/13973e7531c634c4d8659f398940c8c1120e8829/NOTICE.txt

**Comment by mistercrunch** [comment]
This is bad, reverting. FooTable is GPL, we can't use that, let alone making up license headers like we can change license on external libs? 

Plus we already have DataTables in the dependencies.

**Comment by yyyyyyuanfei** [comment]
Hi - this PR uses [FooTable V2](https://github.com/stranbird/airflow/blob/e23a1100c2626f676a994eb1d52a54c6fc58e189/airflow/www/static/footable.js) which is under MIT. The V3 turned to GPL somehow, but we haven't used that.

The discussion about adding MIT license are [here](https://github.com/airbnb/airflow/pull/1422/files#diff-23aa4b370239500d43855f2b3a7b5461).

The goal of introducing FooTable is to use its "RowToggle" [feature](http://fooplugins.com/footable/demos/row-toggle.htm) which is not provided by DataTables. I also tested that there is not conflict between two libraries.

Thanks.

**Comment by yyyyyyuanfei** [comment]
OK, I found that DataTables can also achieve the similar feature: https://datatables.net/examples/api/row_details.html

I'll create a new PR based on that, along with the detail row to show `conf` in the "DagRun" view. How about that? @mistercrunch

**Comment by mistercrunch** [comment]
Hey @stranbird , sorry about the revert, this is a sweet feature that we actually all need!  I got confused when I double-checked the license on FooTable because of the change. I'm currently working on ripping out the Highcharts javascript lib from the project and replacing it with NVD3 because of license incompatibility, so I feel strongly about picking the right libs :)

Can't we just expose the field in the edit view in the CRUD? Note that this means that some secured information will be exposed. Maybe we can let the user decide on `is_extra_encrypted`. If it is set to encrypted we would encrypt and not make it editable (as for password, it need to be fully typed to overwrite), if the user doesn't encrypt they can edit it.
https://github.com/airbnb/airflow/blob/master/airflow/models.py#L453

I think it's just a matter of tuning models.Connection.set_extra and get_extra.

**Comment by r39132** [comment]
@mistercrunch 
Currently, we use fernet key encryption to encrypt certain fields in the DB. We are using FlaskAdmin - which essentially makes all views admin views. As such, an admin should be able to see unencrypted data. Folks who have access to the db but not the admin cannot view the fields, but don't we want people who have access to the admin web ui to have both read and write perms?
